lib/internal/test_runner/runner.js

```diff
@@ -382,7 +381,8 @@ function runTestFile(path, filesWatcher, opts) {
       filesWatcher.runningSubtests.delete(path);
       if (filesWatcher.runningSubtests.size === 0) {
         opts.root.reporter[kEmitMessage]('test:watch:drained');
-        queueMicrotask(() => opts.root.postRun());
+        const postRun = () => opts.root.postRun();
+        PromisePrototypeThen(subTestEnded, postRun, postRun);
```

Comment:
Nit: can be finally? (unless trying to suppress the error)

Reply:
https://github.com/nodejs/node/blob/b0443690f89816ca3b9b8dee7e73f4527b3ff3e0/doc/contributing/primordials.md#L457-L489